### PR TITLE
Some required tag details

### PR DESCRIPTION
**Pull request by liach**
Since I saw FabricMC/fabric#1467, I feel it's necessary to refactor and document some stuff about the required tag lists.

Feel free to ask for more documentation.

cc @LambdAurora

### CODE REVIEW

File: mappings/net/minecraft/tag/RequiredTagList.mapping [hunk 1]
```diff
@@ -25,6 +27,7 @@ CLASS net/minecraft/class_5120 net/minecraft/tag/RequiredTagList
 	METHOD method_33147 addToManager (Lnet/minecraft/class_5415$class_5749;)V
 		ARG 1 manager
 	METHOD method_33148 getRegistryKey ()Lnet/minecraft/class_5321;
+		COMMENT {@return the registry key representing the type of object in this list}
```
Comment (YanisBft) [inline mappings/net/minecraft/tag/RequiredTagList.mapping:30]:
```suggestion
		COMMENT @return the registry key representing the type of object in this list
```

File: mappings/net/minecraft/tag/RequiredTagListRegistry.mapping [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
+	FIELD field_28303 REQUIRED_LIST_KEYS Ljava/util/Set;
 	METHOD method_30196 clearAllTags ()V
 	METHOD method_30198 updateTagManager (Lnet/minecraft/class_5415;)V
 		ARG 0 tagManager
 	METHOD method_30199 (Lnet/minecraft/class_5415;Lnet/minecraft/class_5120;)V
 		ARG 1 list
+	METHOD method_30200 (Lcom/google/common/collect/Multimap;Lnet/minecraft/class_5415;Lnet/minecraft/class_5120;)V
+		ARG 2 list
 	METHOD method_30201 register (Lnet/minecraft/class_5321;Ljava/lang/String;)Lnet/minecraft/class_5120;
+		COMMENT Creates and Registers a required tag list.
```
Comment (YanisBft) [inline mappings/net/minecraft/tag/RequiredTagListRegistry.mapping]:
```suggestion
		COMMENT Creates and registers a required tag list.
```

File: mappings/net/minecraft/tag/RequiredTagList.mapping (hunk 1 above)
Comment (LambdAurora) [reply mappings/net/minecraft/tag/RequiredTagList.mapping:30]: Inline `@return` has special behavior, it generates the brief text too.

Comment (YanisBft) [reply mappings/net/minecraft/tag/RequiredTagList.mapping:30]: alright, didn't know that

Comment (liach) [reply mappings/net/minecraft/tag/RequiredTagList.mapping:30]: Just fyi, 

> [As an inline tag, provides content for the first sentence of a method's description, and a "Returns" section, as if `@return` description were also present. In the default English locale, the first sentence is `Returns` *description* `.`](https://docs.oracle.com/en/java/javase/16/docs/specs/javadoc/doc-comment-spec.html#return)

**Review by LambdAurora — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**